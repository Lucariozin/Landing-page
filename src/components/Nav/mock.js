export default [
  {
    text: 'intro',
    newTab: false,
    to: '#intro',
  },
  {
    text: 'grid-one',
    newTab: false,
    to: '#grid-one',
  },
  {
    text: 'gallery',
    newTab: false,
    to: '#gallery',
  },
  {
    text: 'grid-two',
    newTab: false,
    to: '#grid-two',
  },
  {
    text: 'pricing',
    newTab: false,
    to: '#pricing',
  },
  {
    text: 'contact',
    newTab: false,
    to: '#contact',
  },
];
