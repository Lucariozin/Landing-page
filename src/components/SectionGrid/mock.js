export const textMock = {
  title: 'My grid',
  description: 'Uma breve descrição.',
  backgroundDark: true,
  gridItemsText: [
    {
      title: 'Teste 1',
      content:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      content:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      content:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
};

export const imageMock = {
  title: 'Gallery',
  description: 'Uma breve descrição.',
  gridItemsImages: [
    {
      alt: 'Teste 1',
      src: 'https://source.unsplash.com/random/360x360?r=1',
    },
    {
      alt: 'Teste 2',
      src: 'https://source.unsplash.com/random/360x360?r=2',
    },
    {
      alt: 'Teste 3',
      src: 'https://source.unsplash.com/random/360x360?r=3',
    },
    {
      alt: 'Teste 4',
      src: 'https://source.unsplash.com/random/360x360?r=4',
    },
    {
      alt: 'Teste 5',
      src: 'https://source.unsplash.com/random/360x360?r=5',
    },
    {
      alt: 'Teste 6',
      src: 'https://source.unsplash.com/random/360x360?r=6',
    },
  ],
};
