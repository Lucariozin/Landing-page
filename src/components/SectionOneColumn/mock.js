export default {
  title: 'News coverage and some surprises',
  paragraphs: [
    'The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine’s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.',
    'The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine’s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.',
    'The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine’s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.',
  ],
};
