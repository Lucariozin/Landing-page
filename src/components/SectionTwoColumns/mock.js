export default {
  title: 'January brings us firefox 85',
  description:
    'To wrap up January, we are proud to bring you the release of Firefox 85. In this version we are bringing you support for the :focus-visible pseudo-class in CSS and associated devtools, and the complete removal of Flash support from Firefox.',
  image: {
    src: 'assets/img/javascript.svg',
    alt: 'Uma imagem qualquer',
  },
};
