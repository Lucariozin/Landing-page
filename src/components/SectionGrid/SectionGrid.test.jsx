import { render, screen } from '@testing-library/react';
import { SectionGrid } from '.';
import { textMock, imageMock } from './mock';

describe('<SectionGrid />', () => {
  it('should render <SectionGrid /> with textMock', () => {
    render(<SectionGrid {...textMock} />);

    expect(screen.getByRole('heading', { name: /My grid/i }));
    expect(screen.getByText(/Uma breve descrição./i));
    expect(screen.getAllByText(/Teste/i)).toHaveLength(3);
    expect(screen.getAllByText(/Lorem ipsum dolor sit/i)).toHaveLength(3);
  });

  it('should render <SectionGrid /> with imageMock', () => {
    render(<SectionGrid {...imageMock} />);

    expect(screen.getByRole('heading', { name: /Gallery/i }));
    expect(screen.getByText(/Uma breve descrição./i));
    expect(screen.getAllByAltText(/Teste/i)).toHaveLength(6);
  });

  it('should to match snapshot', () => {
    const { container } = render(<SectionGrid {...textMock} />);
    expect(container).toMatchSnapshot();
  });
});
