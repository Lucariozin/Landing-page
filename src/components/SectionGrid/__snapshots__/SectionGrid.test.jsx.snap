// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionGrid /> should to match snapshot 1`] = `
.c2 {
  padding-bottom: 3rem;
}

.c3 {
  margin-bottom: 1rem;
}

.c6 {
  margin: 0 auto;
  margin-top: 7rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 3rem;
}

.c7 {
  width: 100%;
}

.c9 {
  margin: 0 auto;
  margin-top: 2rem;
  width: 100%;
}

.c0 {
  width: 100%;
  min-height: 100vh;
  padding: 1rem;
  padding-top: 8rem;
  background-image: url(assets/img/background.svg);
  background-position: center center;
  background-size: cover;
  color: var(--white);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
}

.c4 {
  color: var(--white);
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c8 {
  color: var(--white);
  font-size: 3.2rem;
  text-transform: none;
}

.c5 {
  color: var(--white);
  font-style: normal;
  font-size: 2rem;
}

@media (max-width:640px) {
  .c2 {
    text-align: center;
  }
}

@media (max-width:800px) {
  .c6 {
    grid-template-columns: 1fr 1fr;
  }
}

@media (max-width:550px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class=""
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
            >
              My grid
            </h2>
          </div>
          <p
            class="c5"
          >
            Uma breve descrição.
          </p>
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h3
              class="c8"
            >
              Teste 1
            </h3>
            <div
              class="c9"
            >
              <p
                class="c5"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </p>
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c8"
            >
              Teste 2
            </h3>
            <div
              class="c9"
            >
              <p
                class="c5"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </p>
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c8"
            >
              Teste 3
            </h3>
            <div
              class="c9"
            >
              <p
                class="c5"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
